Make all truth tracking collections even with 0 SimHits

bloodyyugo opened the pull request:
I've modified TruthSeedProcessor so that it no longer skips MC events with 0 SimHits in either recoil or tracker collections (issue 1371).  This caused a crash downstream when one of the truth collections was (attempted) to be used.  

In the update, an error-level message will be printed when it encounters an empty SimHits collection but (empty) truth seed and track collections will be made.  

The error messages can be turned off (for instance if the tagger wasn't simulated) by setting the new config variable skip_XXXX to true, for example:   
`truth_tracking.skip_tagger       = True`
a 'skip_recoil' variable exists as well.  Note that these also turn off the relevant truth-track-making code as well.  These config variables are set to 'False' by default. 

I've also changed some log entries from 'info' to 'debug' level.

tomeichlersmith reviewed (changes requested):
- The clang-format check is failing, please format the source files you've changed.[^1]
- Please put the new parameters into the [associated python config class](https://github.com/LDMX-Software/ldmx-sw/blob/8b4650fea34453bd12ccfb7dc03a3645f8616877/Tracking/python/tracking.py#L241-L297). While this isn't necessary (as you've discovered), it is helpful for discoverability of the available run-time parameters.

[^1]: This program is available within the container so you can run `ldmx clang-format -i Tracking/include/Tracking/Reco/TruthSeedProcessor.h Tracking/src/Tracking/Reco/TruthSeedProcessor.cxx` in order to do the formatting. Please let me know if this doesn't work so I don't keep telling people the wrong thing.

bloodyyugo commented:
Ok, I've fixed the formatting and added the variables to tracking.py.  It looks like tracking.py needs some other updating, but I'll add that to my list.

tomeichlersmith commented:
Yea, it looks like you ran the clang-format over the python file. This generally mangles the results unfortunately.

bloodyyugo commented:
> Yea, it looks like you ran the clang-format over the python file. This generally mangles the results unfortunately.

Damn it.   I fixed this up

sophiemiddleton reviewed (approved):
this looks good, thanks for looking at this

tomeichlersmith commented on Tracking/src/Tracking/Reco/TruthSeedProcessor.cxx:
```diff
@@ -57,9 +57,13 @@ void TruthSeedProcessor::configure(framework::config::Parameters& parameters) {
       "inflate_factors", {10., 10., 10., 10., 10., 10.});
 
-  // In tracking frame
+  // In tracking frame:  MG where do these numbers come from?
```
These numbers come from an approximation of where the beam generator places the beam (without smearing) just upstream of all detector components. `-880.1` is just upstream of TriggerPad1 in `z` and `-44` is approximately the bend in `x` imposed on the beam by the magnet (I think 4GeV but could be 8GeV). Then they are re-ordered to be in tracking coordinates.

tomeichlersmith reviewed (approved):
Looking good, thanks for cleaning things up!
